### PR TITLE
hack/scripts-dev: add "docker-dns-example-certs-common-name-run"

### PR DESCRIPTION
**Pull request by gyuho**
Address https://github.com/coreos/etcd/issues/8502.
Separate out from https://github.com/coreos/etcd/pull/8954.

/cc @fanminshi

### CODE REVIEW

File: hack/scripts-dev/docker-dns/certs-common-name/run.sh [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+  put abc def
+
+ETCDCTL_API=3 ./etcdctl \
+  --cacert=/certs-common-name/ca.crt \
+  --cert=/certs-common-name/server.crt \
+  --key=/certs-common-name/server.key.insecure \
+  --endpoints=https://m1.etcd.local:2379,https://m2.etcd.local:22379,https://m3.etcd.local:32379 \
+  get abc
+
+sleep 1s && printf "\n"
```
Comment (fanminshi) [inline hack/scripts-dev/docker-dns/certs-common-name/run.sh]: no sleep?
same as many places down below/

File: hack/scripts-dev/docker-dns/certs-common-name/run.sh [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+echo "Step 1. creating root role"
```
Comment (fanminshi) [inline hack/scripts-dev/docker-dns/certs-common-name/run.sh]: I don't quite understand this test.
why are we creating root user and root role?

File: hack/scripts-dev/docker-dns/certs-common-name/run.sh (hunk 1 above)
Comment (gyuho) [reply hack/scripts-dev/docker-dns/certs-common-name/run.sh]: Let's keep sleep here. I am trying to put example workflow here with print statements. Otherwise, it's hard to follow.

File: hack/scripts-dev/docker-dns/certs-common-name/run.sh (hunk 2 above)
Comment (gyuho) [reply hack/scripts-dev/docker-dns/certs-common-name/run.sh]: User must create `root` user and role first. It's special user and role. https://github.com/coreos/etcd/blob/master/Documentation/op-guide/authentication.md#special-users-and-roles

Comment (fanminshi) [reply hack/scripts-dev/docker-dns/certs-common-name/run.sh]: you are testing this scenario right?https://github.com/coreos/etcd/blob/master/Documentation/op-guide/authentication.md#using-tls-common-name

Comment (gyuho) [reply hack/scripts-dev/docker-dns/certs-common-name/run.sh]: Yes.

**Comment by fanminshi** [comment]
this script is more a demo of how to use common name than a test script. suggest rename this as a demo script.
